pkg/controllers/migration/virtualmachine.go

```diff
@@ -199,24 +204,38 @@ func (h *virtualMachineHandler) OnVirtualMachineChange(_ string, vmObj *migratio
 			"namespace":               vm.Namespace,
 			"spec.virtualMachineName": vm.Spec.VirtualMachineName,
 		}).Info("The VM was imported successfully")
-		return vm, h.tidyUpObjects(vm)
+		return nil, h.tidyUpObjects(vm)
 	case migration.VirtualMachineImportInvalid:
 		logrus.WithFields(logrus.Fields{
 			"name":                    vm.Name,
 			"namespace":               vm.Namespace,
 			"spec.virtualMachineName": vm.Spec.VirtualMachineName,
 		}).Error("The VM import spec is invalid")
-		return vm, nil
+		return nil, nil
 	case migration.VirtualMachineMigrationFailed:
 		logrus.WithFields(logrus.Fields{
 			"name":                    vm.Name,
 			"namespace":               vm.Namespace,
 			"spec.virtualMachineName": vm.Spec.VirtualMachineName,
 		}).Error("The VM import has failed")
-		return vm, nil
+		return nil, h.triggerCleanup(vm)
+	}
+
+	return nil, nil
+}
+
+func (h *virtualMachineHandler) OnVirtualMachineRemove(_ string, vmi *migration.VirtualMachineImport) (*migration.VirtualMachineImport, error) {
+	if vmi == nil {
+		return nil, nil
 	}
 
-	return vm, nil
+	// Make sure all temporary files are really cleaned up.
+	err := h.triggerCleanup(vmi)
+	if err != nil {
```

Comment:
@ibrokethecloud @FrankYang0529 Perhaps we should ignore any errors that occur here to avoid getting caught in a reconciliation loop. If files in /tmp/vm-import-controller are not deleted, it is less serious than if the deletion of a VirtualMachineImport CR cannot be completed as a result. What do you think?

Reply:
i agree.. we should log error and continue, else wrangler will not remove finalizer and vm import object will be stuck